### ISSUE TITLE
On usage of `commonIdea`

### ISSUE THREAD

**Issue by JacquesCarette** [JacquesCarette/Drasil#3315]
`commonIdea` and `commonIdeaDict` should check if the list they are given is empty and, if so, complain. `mkIdea` should be used instead.

There are quite a few abuses of this in Drasil, and they should all be fixed.

**Comment by hrzhuang** [comment]
Is there a reason why `AuxConsProg` uses a `CI` instead of a type class restriction like `NamedIdea`?

**Comment by JacquesCarette** [comment]
Probably no good reason for it. Try to fix it and see what happens!

**Comment by hrzhuang** [comment]
So the problem here is that `AuxConsProg` constructs a `AuxConstntSec`, which is used in `DocSection`. If we use a type class like `NamedIdea` for the first argument of `AuxConsProg`, then `AuxConstntSec` would need a type parameter, which in turn would require `DocSection` to have a type parameter, and this seems to be a change that would require some more deliberation.

The `CI` argument to `AuxConsProg` seems to just be used for the name of the program (e.g. GlassBR). Does it in fact make sense to use a raw `CI` here?

**Comment by hrzhuang** [comment]
Another "abuse" of `commonIdea` is the smart constructors `dcc'` and `dccWDS'` for `CommonConcept`. I suppose if `CI`s should always have a non-empty list of domains, then that list of domains should also be added as an argument to `dcc'` and `dccWDS'`?

**Comment by JacquesCarette** [comment]
Right - things like `DocSection` should have much more concrete contents. Having it have an extra type parameter seems like it could spiral out of control (though it wouldn't be wrong).

Your follow-up question is excellent: we need a name, not a 'concept' here. The only reason we'd want a concept is to be able to cross-reference the UID somehow. Which might happen!

As to: what is the 'domain' of a program name? This, I don't know. Which maybe is a further hint that this isn't a "concept" per se.

There is a bigger issue with these representations - I can't find the number right now. @balacij ?

**Comment by balacij** [comment]
Sorry, I'm not entirely sure what the larger issue you're referring to is, @JacquesCarette. Is it about the "gradual chunk building?"

**Comment by balacij** [comment]
Gotcha, that's #2873. We also have a larger discussion in [chapter 8 of my master's thesis](https://github.com/JacquesCarette/Drasil/files/11432344/thesis.pdf)
.

**Comment by daijingz** [comment]
tomorrow working

**Comment by balacij** [comment]
No, they shouldn't be combined together, and we are working on redesigning our chunks right now, so I would stay away from working on that stuff right now.
